app/src/main/java/bg/zahov/app/ui/settings/SettingsFragment.kt

```diff
@@ -79,133 +51,25 @@ class SettingsFragment : Fragment() {
                 }
             }
         })
-
-        binding.apply {
-            signOutButton.setOnClickListener {
-                settingsViewModel.logout()
-            }
-            editProfile.setOnClickListener {
-                findNavController().navigate(R.id.settings_to_edit_profile)
-            }
-            github.setOnClickListener {
-                openLink("https://github.com/HauntedMilkshake/fitness_app")
-            }
-            bugReport.setOnClickListener {
-                openLink("https://github.com/HauntedMilkshake/fitness_app/issues")
-            }
-            deleteAccount.setOnClickListener {
-                settingsViewModel.deleteAccount()
-            }
-
-        }
-        initDefaultSettingsViews()
-    }
-
-    private fun initRadioSettingsView(
-        view: RadioGroupSettingsView,
-        title: String,
-        radioOptions: List<String>,
-        settings: Settings,
-    ) {
-        view.initViewInformation(title, radioOptions, settings)
-        view.settingsChangeListener = object : SettingsChangeListener {
-            override fun onSettingChanged(title: String, newValue: Any) {
-                settingsViewModel.writeNewSetting(title, newValue)
-            }
-        }
-    }
-
-    private fun initSwitchSettingsView(
-        view: SwitchSettingsView,
-        title: String,
-        subtitle: String,
-        settings: Settings,
-    ) {
-        view.initViewInformation(title, subtitle, settings)
-        view.settingsChangeListener = object : SettingsChangeListener {
-            override fun onSettingChanged(title: String, newValue: Any) {
-                settingsViewModel.writeNewSetting(title, newValue)
+        return ComposeView(requireContext()).apply {
+            requireActivity().hideBottomNav()
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                SettingsScreen(
+                    settingsViewModel,
+                    navigateBack = { findNavController().navigate(R.id.settings_to_welcome) },
+                    navigateEditProfile = {
+                        findNavController().navigate(R.id.settings_to_edit_profile)
```

Comment:
Use one type of formatting - one line or multiple lines for callbacks

Reply:
this was done from the inbuild formatting